optimum/neuron/modeling_base.py

```diff
@@ -484,16 +493,21 @@ def _pad_to_compiled_shape(self, inputs: Dict[str, "torch.Tensor"]):
 
             # Pad to batch size: dimension 0 (pad_token_id can't be 0)
             padding = (0,) * len(padding)
-            if self.neuron_config.dynamic_batch_size is True and input_tensor.size(0) % target_shapes[0] == 0:
+            is_encoder_decoder = getattr(self.config, "is_encoder_decoder", False)
+            if (
+                not is_encoder_decoder
+                and self.neuron_config.dynamic_batch_size is True
```

Comment:
I really dont see the coherence here, why not `not is_encoder_decoder is True and self.neuron_config.dynamic_batch_size is True`?
I think it's an overkill to check for `is True` but that's my opinion.